src/ars/core/repository.py

```diff
@@ -311,6 +311,7 @@ async def delete_dataset(self, dataset_id: str) -> None:
         Raises a `DatasetNotFoundError` if the dataset was not found.
         """
         try:
+            await self._dataset_dao.delete(id_=dataset_id)
             async for request in self._request_dao.find_all(
```

Comment:
The `find_all` does not raise ResourceNotFoundError, so you can move the loop after the try block, or into an else block.

Reply:
Thanks for the feedback, I just moved it after the try block. 
Shouldn't make much of a difference if it's in the else branch or just after the try block.